packages/zosfiles/src/methods/download/Download.ts

```diff
@@ -271,18 +283,29 @@ export class Download {
                     extension = extension.replace(/^\.+/g, "");
                 }
 
+                const memberFilePath = posix.join(baseDir, fileName + IO.normalizeExtension(extension));
```

Comment:
We should use the platform-specific implementation of path instead of `posix`. The current logic assumes that the path is in POSIX format, which may have unintended side effects for Windows clients.